modules/components/pages/inputs/ockam_kafka.adoc

```diff
@@ -0,0 +1,635 @@
+
+The secret for the AWS credentials in use.
+
+include::components:partial$secret_warning.adoc[]
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].aws.credentials.token`
+
+The token for the AWS credentials in use. This is a required value for short-term credentials.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].aws.credentials.from_ec2_role`
+
+Use the credentials of a host EC2 machine configured to assume an https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html[IAM role associated with the instance^].
+
+*Type*: `bool`
+
+*Default*: `false`
+
+Requires version 4.2.0 or newer
+
+=== `kafka.sasl[].aws.credentials.role`
+
+The role ARN to assume.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].aws.credentials.role_external_id`
+
+An external ID to use when assuming a role.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.multi_header`
+
+Decode headers into lists to allow the handling of multiple values with the same key.
+
+*Type*: `bool`
+
+*Default*: `false`
+
+=== `kafka.batching`
+
+Configure a xref:configuration:batching.adoc[batching policy] for individual topic partitions. This allows the input to batch messages together before flushing them for processing. Batching may improve performance and is useful for windowed processing as it preserves the ordering of topic partitions.
+
+*Type*: `object`
+```yml
+# Examples
+batching:
+  byte_size: 5000
+  count: 0
+  period: 1s
+batching:
+  count: 10
+  period: 1s
+batching:
+  check: this.contains("END BATCH")
+  count: 0
+  period: 1m
+```
+=== `kafka.batching.count`
+
+The number of messages after which the batch is flushed. Set to `0` to disable count-based batching.
+
+*Type*: `int`
+
+*Default*: `0`
+
+=== `kafka.batching.byte_size`
+
+The amount of bytes at which the batch is flushed. Set to `0` to disable size-based batching.
+
+*Type*: `int`
+
+*Default*: `0`
+
+=== `kafka.batching.period`
+
+The period of time after which an incomplete batch is flushed regardless of its size.
+
+*Type*: `string`
+
+*Default*: `""`
+
+```yml
+# Examples
+period: 1s
+period: 1m
+period: 500ms
+```
+=== `kafka.batching.check`
+
+A xref:guides:bloblang/about.adoc[Bloblang query] that returns a boolean value indicating whether a message should end a batch.
+
+*Type*: `string`
+
+*Default*: `""`
+
+```yml
+# Examples
+check: this.type == "end_of_transaction"
+```
+
+=== `kafka.batching.processors`
+
+For aggregating and archiving message batches, you can add a list of xref:components:processors/about.adoc[processors] to apply to a batch as it is flushed (optional). All resulting messages are flushed as a single batch even when you configure processors to split the batch into smaller batches.
+
+*Type*: `array`
+
+```yml
+# Examples
+processors:
+  - archive:
+      format: concatenate
+processors:
+  - archive:
+      format: lines
+processors:
+  - archive:
+      format: json_array
+```
+
+=== `disable_content_encryption`
```

Comment:
Missing descriptions for all fields from here to the end of the page. Could you supply or give a source?

Reply:
Added with help from Ockam.